### DISCUSSION TITLE
Реализация примера по распознаванию документов с использованием OCR и ElasticSearch (ReportsOcrAndSearch)

### DISCUSSION THREAD

**Discussion by AntonIV8** [Flexberry/rfcs#32] [General]
В рамках итерации необходимо создать тестовый стенд, демонстрирующий распознавание pdf документов по технологии OCR и сохранение их в текстовых файлах и поиск по содержимому документов.

Пользователю будет доступен интерфейс стандартного ember-приложения. Объектом данных будет выступать сущность Report, у которой будет один параметр - файл отчета.
Пользователь может загружать отчеты в формате pdf.
При загрузке файла, происходит его распознавание и сохранение в виде текстового файла. Также при этом происходит загрузка файла в индексную БД ElasticSearch.

В ember-приложении пользователю будет доступна функция поиска по тексту.  В результате поиска будет выдан список имен файлов-отчетов, в содержании которого текст был найден.

Схема стенда:

![image](https://github.com/Flexberry/rfcs/assets/13151962/16e1525e-96ec-46b6-a877-c1e784e102b8)

Репозиторий: https://github.com/Flexberry/Flexberry.ReportsOcrAndSearch.Sample
Ветка: main

**Comment by kafnevod** [comment]
Весь вопрос какие PDF-файлы.

Если созданные в редакторе, то они довольно легко конвертируются различными open-source программами.

Распознавание по технологии OCR неободимо только  для PDF-файлов, хранящих сканированные страницы

Мы будем поддерживать ТОЛЬКО второй вариант PDF?

Или оба?

**Reply by bratchikov** [reply]
Только сканы.
